plugins/modules/kutils.py

```diff
@@ -0,0 +1,351 @@
+      - If O(keytab_path) is not specified will try to use default client keytab path (C(-i) option).
+    type: bool
+  keytab_path:
+    description:
+      - Use when O(keytab=True) to specify path to a keytab file.
+      - It is required to specify O(password) or O(keytab_path).
+    type: path
+requirements:
+  - krb5-user and krb5-config packages
+extends_documentation_fragment:
+  - community.general.attributes
+'''
+
+EXAMPLES = r'''
+- name: Get kerberos ticket using default principal
+  community.general.kutils:
+    password: some_password
+
+- name: Get kerberos ticket using keytab
+  community.general.kutils:
+    keytab: true
+    keytab_path: /etc/ipa/file.keytab
+
+- name: Get kerberos ticket using principal name
+  community.general.kutils:
+    password: some_password
+    principal: admin
+
+- name: Get kerberos ticket using principal with realm
+  community.general.kutils:
+    password: some_password
+    principal: admin@IPA.TEST
+
+- name: Check for existence by ticket cache
+  community.general.kutils:
+    cache_name: KEYRING:persistent:0:0
+
+- name: Make sure default ticket is destroyed
+  community.general.kutils:
+    state: absent
+
+- name: Make sure specific ticket destroyed by principal
+  community.general.kutils:
+    state: absent
+    principal: admin@IPA.TEST
+
+- name: Make sure specific ticket destroyed by cache_name
+  community.general.kutils:
+    state: absent
+    cache_name: KEYRING:persistent:0:0
+
+- name: Make sure all tickets are destroyed
+  community.general.kutils:
+    state: absent
+    kdestroy_all: true
+'''
+
+from ansible.module_utils.basic import AnsibleModule, env_fallback
+from ansible_collections.community.general.plugins.module_utils.cmd_runner import CmdRunner, cmd_runner_fmt
+
+
+class IPAKeytab(object):
+    def __init__(self, module, **kwargs):
+        self.module = module
+        self.password = kwargs['password']
+        self.principal = kwargs['principal']
+        self.state = kwargs['state']
+        self.kdestroy_all = kwargs['kdestroy_all']
+        self.cache_name = kwargs['cache_name']
+        self.start_time = kwargs['start_time']
+        self.renewable = kwargs['renewable']
+        self.forwardable = kwargs['forwardable']
+        self.proxiable = kwargs['proxiable']
+        self.address_restricted = kwargs['address_restricted']
+        self.canonicalization = kwargs['canonicalization']
+        self.enterprise = kwargs['enterprise']
+        self.renewal = kwargs['renewal']
+        self.validate = kwargs['validate']
+        self.keytab = kwargs['keytab']
+        self.keytab_path = kwargs['keytab_path']
+
+        self.kinit_executable = [module.get_bin_path('kinit', True)]
+        self.kdestroy_executable = [module.get_bin_path('kdestroy', True)]
+        self.klist_executable = [module.get_bin_path('klist', True)]
+
+        self.kinit = CmdRunner(
+            module,
+            command=self.kinit_executable,
```

Comment:
`CmdRunner` will perform the `get_bin_path()` step for you automatically. Again, something I should add to the documentation - thanks for using it and exposing these shortcomings in the docs. ;-)

Reply:
that would be handy, thanks :)